Makefile

```
@@ -1,54 +1,76 @@
+check/updates: TFLINT_LATEST=$(shell curl -s "https://api.github.com/repos/terraform-linters/tflint/releases/latest" | jq -r -M '.tag_name' | sed -e 's/^v//')
+check/updates: PACKER_LATEST=$(shell curl -s https://checkpoint-api.hashicorp.com/v1/check/packer | jq -r -M '.current_version')
+check/updates: TERRAFORM_LATEST = $(shell curl -s https://checkpoint-api.hashicorp.com/v1/check/terraform | jq -r -M '.current_version')
+check/updates:
+	@if [ "${TERRAFORM_VERSION}" != "${TERRAFORM_LATEST}" ] ; then \
+	  echo "${RED}terraform ${TERRAFORM_VERSION}${YELLOW} - update available to terraform ${TERRAFORM_LATEST}${RESET}" ; \
+	else \
+	  echo "${GREEN}terraform ${TERRAFORM_VERSION} - terraform is up to date.${RESET}" ; \
+	fi
+	@if [ "${PACKER_VERSION}" != "${PACKER_LATEST}" ] ; then \
+	  echo "${RED}packer ${PACKER_VERSION}${YELLOW} - update available to packer ${PACKER_LATEST}${RESET}" ; \
+	else \
+	  echo "${GREEN}packer ${PACKER_VERSION} - packer is up to date.${RESET}" ; \
+	fi
+	@if [ "${TFLINT_VERSION}" != "${TFLINT_LATEST}" ] ; then \
+	  echo "${RED}tflint ${TFLINT_VERSION}${YELLOW} - update available to tflint ${TFLINT_LATEST}${RESET}" ; \
+	else \
+	  echo "${GREEN}tflint ${TFLINT_VERSION} - tflint is up to date.${RESET}" ; \
+	fi
+	@if [ "${PRECOMMIT_VERSION}" != "${PRECOMMIT_LATEST}" ] ; then \
+	  echo "${RED}pre-commit ${PRECOMMIT_VERSION}${YELLOW} - update available to pre-commit ${PRECOMMIT_LATEST}${RESET}" ; \
+	else \
+	  echo "${GREEN}pre-commit ${PRECOMMIT_VERSION} - pre-commit is up to date.${RESET}" ; \
+	fi
+
 .PHONY: docker/build
 ## Build the docker image
 docker/build:
-	docker build -t ${DOCKER_HUB_REPO}:${DOCKER_IMAGE_TAG} .
+	docker build \
+	  --build-arg TERRAFORM_VERSION=${TERRAFORM_VERSION} \
+	  --build-arg PACKER_VERSION=${PACKER_VERSION} \
+	  --build-arg TFLINT_VERSION=${TFLINT_VERSION} \
+	  --build-arg PRECOMMIT_VERSION=${PRECOMMIT_VERSION} \
+	  -t ${DOCKER_IMAGE} . | cat
```

for what reason did you pipe the output of each target to cat? Won't this suppress any potential error code that might occur inside the chain of commands?

---

true... hmm... let me think about other ways...

---

you can just get rid of the cat or for what would you use it in the first place?

---

fixed. good catch ;) thanks..